### ISSUE TITLE
Feature Request: Resource for Managing Glue Security Configurations

### ISSUE THREAD

**Issue by bflad** [terraform-providers/terraform-provider-aws#6286]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

The AWS Glue service added security configurations as a method for handling encryption options for various Glue resources.

### New or Affected Resource(s)

* `aws_glue_security_configuration`

### Potential Terraform Configuration

```hcl
# New Resource, not implemented and details may change during implementation
resource "aws_glue_security_configuration" "example" {
  name = ""

  encryption_configuration {
    cloudwatch_encryption {
      cloudwatch_encryption_mode = ""
      kms_key_arn = ""
    }

    job_bookmarks_encryption {
      job_bookmarks_encryption_mode = ""
      kms_key_arn = ""
    }

    s3_encryption {
      kms_key_arn = ""
      s3_encryption_mode = ""
    }
  }
}
```

### References

* https://docs.aws.amazon.com/sdk-for-go/api/service/glue/#Glue.CreateSecurityConfiguration
* https://docs.aws.amazon.com/glue/latest/dg/encryption-security-configuration.html

**Comment by bflad** [comment]
Pull request submitted: #6288

**Comment by bflad** [comment]
This has been released in [version 1.42.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1420-october-31-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
